### PR TITLE
Create test for US recurring contributions (S&C test 3 - stars & pikes)

### PR DESCRIPTION
**Pull request by JustinPinner**
## What does this change?
Prepare for US recurring contribution test.

## Does this affect other platforms - Amp, Apps, etc?
No
## Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?
No

####  Control

| Channel | Campaign Code | Goes to |
|---------|---------------|------------|
|**Header Become Supporter (Desktop)**|gdnwb_copts_memco_sandc_us_rec_cont_control_header_become_supporter |`membership-frontend`|
|**Header Make a Contribution (Tablet and Mobile)**|gdnwb_copts_memco_sandc_us_rec_cont_control_header_make_contribution|`contributions-frontend`|
|**Side Bar - Become Supporter**|gdnwb_copts_memco_sandc_us_rec_cont_control_side_menu_become_supporter|`membership-frontend`|
| **Engagement Banner** |gdnwb_copts_memco_sandc_us_rec_cont_control_banner |`contributions-frontend`|
| **Epic**  |gdnwb_copts_memco_sandc_support_us_rec_cont_control_epic|Contribution button to `contribute-frontend` and member button to `membership-frontend`|
| **Live Blog** |gdnwb_copts_memco_sandc_support_us_rec_cont_control_epic_liveblog|Contribution button to `contribute-frontend` and member button to `membership-frontend`|

#### Variant Support

| Channel | Campaign Code | Context |
|---------|---------------|-----------|
|**Header Become Supporter (Desktop)**|gdnwb_copts_memco_sandc_us_rec_cont_support_header_become_supporter|true|
|**Header Make a Contribution (Tablet and Mobile)**|gdnwb_copts_memco_sandc_us_rec_cont_support_header_make_contribution|true|
|**Side Bar - Become Supporter**|gdnwb_copts_memco_sandc_us_rec_cont_support_side_menu_become_supporter | true|
| **Engagement Banner** | gdnwb_copts_memco_sandc_us_rec_cont_support_banner | true |
| **Epic**  |gdnwb_copts_memco_sandc_support_us_rec_cont_support_epic|Not Present|
| **Liveblog** | gdnwb_copts_memco_sandc_us_rec_cont_support_liveblog |Not Present| 

### Header and engagement banner (in-test):
supportfrontendusheader engagementbanner-intest

### Menu (in-test):
supportfrontendusmenu-intest

### Epic (in-test):
supportfrontendusepic-intest

### Header and engagement banner (in-control):
banner-control

### Menu (in-control):
sidemenu-control

### Epic (in-control)
epic-control

## Tested in CODE?
Yes

### CODE REVIEW

**Comment by svillafe** [comment]
cc: @guardian/contributions

**Review by Ap0c — COMMENTED** [review]
Looks ok to me.

File: static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-us-recurring-contributions.js [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+            <a href="#" itemprop="url" class="block-time__link">
+                <time data-relativeformat="med" itemprop="datePublished" class="js-timestamp"></time>
+                <span class="block-time__absolute"></span>
+            </a>
+        </p>
+        <div class="block-elements block-elements--no-byline">
+            <p>
+                <em>
+                    Since you’re here &hellip; we have a small favour to ask. More people are reading the Guardian than ever but advertising revenues across the media are falling fast. And <span class="contributions__highlight">unlike many news organisations, we haven’t put up a paywall &ndash; we want to keep our journalism as open as we can</span>. So you can see why we need to ask for your help. The Guardian’s independent, investigative journalism takes a lot of time, money and hard work to produce. But we do it because we believe our perspective matters &ndash; because it might well be your perspective, too.
+                </em>
```
Comment (jranks123) [inline static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-us-recurring-contributions.js]: Do we need to have the template and copy in the test? We have the Acquisitions copy module for storying copy, and the acquisitions-epic-livegblog.html template, you should be able to implement this using those

**Review by jranks123 — DISMISSED** [review]

File: static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-us-recurring-contributions.js (hunk 1 above)
Comment (jranks123) [reply static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-us-recurring-contributions.js]: You can see an example here: 

If you look in acquisitions-epic-liveblog.js, you will see in the control variant the following: 

```template(variant) {
                    return template(liveblogEpicTemplate, {
                        copy: liveblogCopy(
                            variant.options.membershipURL,
                            variant.options.contributeURL
                        ),
                        componentName: variant.options.componentName,
                    });
                },```

Comment (JustinPinner) [reply static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-us-recurring-contributions.js]: Changing this doesn't make this test any better / more robust etc. If anything, having it set here protects it from other unintentional changes to the copy made for other reasons.

Comment (jranks123) [reply static/src/javascripts/projects/common/modules/experiments/tests/acquisitions-support-us-recurring-contributions.js]: The point of having the templates and copy module is to reduce mistakes, ensure consistency across tests and clean up the test templates. 

While it doesn't make the test "better", it keeps everything tidy and means that test files are less cluttered. By putting the template in here, you are needlessly making the test file bloated and duplicating code. 

Also, above here the button template code is written into the test file, when there is already a template in the project (acquisitions-epic-single-button.html) which would do the job.

**Comment by JustinPinner** [comment]
Discussed with @jranks123 and he's happy to put this out as-is on the grounds that we can refactor the template changes after the test is live.

**Review by jranks123 — APPROVED** [review]
👍  Happy for this to go through, and for the template refactoring to happen for the next test. 

Suggestion: write a selenium test that can simulate a multichannel test (with clicks to all the different links) that can be ran after each change. The barrier to making changes is too high at the moment as the testing is so tricky, which will lead to compromises when wanting to move at speed

**Pull request merged**